src/request/types/index.ts

```diff
@@ -0,0 +1,43 @@
+import { RpcSuccessResponse } from 'src/types';
+import { GetAddresses, GetInfo, SendTransfer, SignMessage, SignPsbt } from './btcMethods';
+import {
+  StxCallContract,
+  StxDeployContract,
+  StxGetAccounts,
+  StxGetAddresses,
+  StxSignStructuredMessage,
+  StxSignStxMessage,
+  StxSignTransaction,
+  StxTransferStx,
+} from './stxMethods';
+
+export interface StxRequests {
+  stx_callContract: StxCallContract;
+  stx_deployContract: StxDeployContract;
+  stx_getAccounts: StxGetAccounts;
+  stx_getAddresses: StxGetAddresses;
+  stx_signMessage: StxSignStxMessage;
+  stx_signStructuredMessage: StxSignStructuredMessage;
+  stx_signTransaction: StxSignTransaction;
+  stx_transferStx: StxTransferStx;
+}
+
+export type StxRequestMethod = keyof StxRequests;
+
+export interface BtcRequests {
+  getInfo: GetInfo;
+  getAddresses: GetAddresses;
+  signMessage: SignMessage;
+  sendTransfer: SendTransfer;
+  signPsbt: SignPsbt;
+}
+
+export type BtcRequestMethod = keyof BtcRequests;
+
+export type Requests = BtcRequests & StxRequests;
+
+export type Return<Method> = Method extends keyof Requests ? Requests[Method]['result'] : never;
```

Comment:
what's the difference between this and:

```suggestion
export type Return<Method extends keyof Requests> = Requests[Method]['result'];
```

Reply:
True, yours is better as you'll get a type warning directly rather than silently accepting the method name with a never result.